clang-tools-extra/clangd/ParsedAST.cpp

```diff
@@ -451,6 +451,7 @@ ParsedAST::build(llvm::StringRef Filename, const ParseInputs &Inputs,
   DiagnosticConsumer *DiagConsumer = &ASTDiags;
   IgnoreDiagnostics DropDiags;
   if (Preamble) {
+    CI->TargetOpts = Preamble->TargetOpts;
```

Comment:
It'd be useful to add a comment here explaining why we need it (i.e. "workaround a crash, see ParsedASTTest.PreambleWithDifferentTarget")